include/RAJA/policy/cuda/kernel/Sync.hpp

```diff
@@ -43,12 +43,16 @@ namespace statement
  * A RAJA::kernel statement that performs a CUDA __syncthreads().
  */
 struct CudaSyncThreads : public internal::Statement<camp::nil> {
+  private:
+  CudaSyncThreads() = default;
```

Comment:
Instead of deleting this constructor, delete the constructor in internal::Statement.
That will force the issue for all Statements

Reply:
Done. Still simply gives a warning in gcc though.